pkg/upgrade/deploy.go

```diff
@@ -17,6 +22,11 @@ func (u *simpleUpgrader) Deploy(ctx context.Context, cs *api.OpenShiftManagedClu
 		return err
 	}
 
+	err = managedcluster.WaitForHealthz(ctx, cs.Config.AdminKubeconfig)
```

Comment:
All of these health checks seems to be all over the place. We should probably consolidate this api server healthcheck, the node healthchecks below, and the infra services healthcheck inside the healthcheck plugin function.

Reply:
Separate issue is fine if it makes sense to do this.

Reply:
https://github.com/openshift/openshift-azure/issues/675